src/SDK/Resource/ResourceInfo.php

```diff
@@ -69,10 +73,34 @@ public static function defaultResource(): self
                 ResourceConstants::TELEMETRY_SDK_NAME => 'opentelemetry',
                 ResourceConstants::TELEMETRY_SDK_LANGUAGE => 'php',
                 ResourceConstants::TELEMETRY_SDK_VERSION => 'dev',
+                ResourceConstants::SERVICE_NAME => 'unknown_service',
             ]
         ));
     }
 
+    /**
+     * Create resource attributes from OTEL_RESOURCE_ATTRIBUTES key=value comma-separated entries
+     * @see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#specifying-resource-information-via-an-environment-variable
+     */
+    public static function environmentResource(): self
+    {
+        $attributes = [];
+        $string = getenv('OTEL_RESOURCE_ATTRIBUTES');
+        if ($string && false !== strpos($string, '=')) {
+            foreach (explode(',', $string) as $pair) {
+                list($key, $value) = explode('=', $pair);
+                $attributes[trim($key)] = trim($value);
+            }
+        }
+        //@see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration
+        $serviceName = getenv('OTEL_SERVICE_NAME');
+        if ($serviceName) {
```

Comment:
I agree that this could be a warning. I had started adding some logging in this branch for situations like this (for the eventual implementation of "must not throw exceptions"), but decided to keep this PR on-topic and not do it yet. It's amongst the next couple of things I want to work on :)

Reply:
Ok great!